client/deployments/client.go

```diff
@@ -390,3 +396,202 @@ func (c *Client) DeleteArtifact(
+	}
+	req.Header.Set("Authorization", "Bearer "+string(token))
+
+	reqDump, _ := httputil.DumpRequest(req, false)
+	log.Verbf("sending request: \n%v", string(reqDump))
+
+	rsp, err := c.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "GET /artifacts request failed")
+	}
+	defer rsp.Body.Close()
+
+	rspDump, _ := httputil.DumpResponse(rsp, true)
+	log.Verbf("response: \n%v\n", string(rspDump))
+
+	body, err := io.ReadAll(rsp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "GET /artifacts request failed")
+	}
+
+	var artifact Artifact
+	err = json.Unmarshal(body, &artifact)
+	if err != nil {
+		return nil, errors.Wrap(err, "GET /artifacts request failed")
+	}
+
+	return &artifact, nil
+}
+
+func (c *Client) getLink(
+	artifactID, token string,
+) (*DownloadLink, error) {
+	req, err := http.NewRequest(http.MethodGet, strings.ReplaceAll(c.artifactDownloadURL, ":id", artifactID), nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "Cannot create request")
+	}
+	req.Header.Set("Authorization", "Bearer "+string(token))
+
+	reqDump, _ := httputil.DumpRequest(req, false)
+	log.Verbf("sending request: \n%v", string(reqDump))
+
+	rsp, err := c.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "GET /artifacts request failed")
+	}
+	defer rsp.Body.Close()
+
+	rspDump, _ := httputil.DumpResponse(rsp, true)
+	log.Verbf("response: \n%v\n", string(rspDump))
+
+	body, err := io.ReadAll(rsp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "GET /artifacts request failed")
+	}
+
+	var link DownloadLink
+	err = json.Unmarshal(body, &link)
+	if err != nil {
+		return nil, errors.Wrap(err, "GET /artifacts request failed")
+	}
+
+	return &link, nil
+}
+
+func (c *Client) downloadFile(size int64, localFileName string, resp *http.Response, noProgress bool) error {
+	path := resp.Header.Get("X-MEN-FILE-PATH")
+	uid := resp.Header.Get("X-MEN-FILE-UID")
+	gid := resp.Header.Get("X-MEN-FILE-GID")
+	var n int64
+
+	file, err := os.Create(localFileName)
+	if err != nil {
+		return errors.Wrap(err, "Cannot create file")
+	}
+	defer file.Close()
+	if err != nil {
+		log.Errf("Failed to create the file %s locally\n", path)
+		return err
+	}
+
+	if resp.Header.Get("Content-Type") != "application/vnd.mender-artifact" {
+		return fmt.Errorf("Unexpected Content-Type header: %s", resp.Header.Get("Content-Type"))
+	}
+	if err != nil {
+		log.Err("downloadFile: Failed to parse the Content-Type header")
+		return err
+	}
+	i, _ := strconv.Atoi(resp.Header.Get("Content-Length"))
+	sourceSize := int64(i)
+	source := resp.Body
+
+	if !noProgress {
+		var bar *pb.ProgressBar
+		bar = pb.New64(sourceSize).
+			Set(pb.Bytes, true).
+			SetRefreshRate(time.Millisecond * 100)
+		bar.Start()
+		// create proxy reader
+		reader := bar.NewProxyReader(source)
+		n, err = io.Copy(file, reader)
+	} else {
+		n, err = io.Copy(file, source)
+	}
+	if err != nil {
+		log.Verbf("wrote: %d\n", n)
+		return err
+	}
+	log.Verbf("wrote: %d\n", n)
```

Comment:
Nitpick: Could move the `log` before the `if`, since you do the same in both cases.

Reply:
done in both places :+1: